File: src/exercises/combination_sum_4/go/solution.go
func combinationSum4(nums []int, target int) int {
	sort.Ints(nums)
	dp := make(map[int]int)
	dp[target] = 1
	for total := target; total > 0; total-- {
		for _, num := range nums {
			if total < num {
				break
			}
			dp[total-num] += dp[total]
		}
	}
	return dp[0]
}

File: src/exercises/combination_sum_4/python/solution.py
class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        nums.sort()
        dp = defaultdict(int)
        dp[target] = 1
        for total in range(target, 0, -1):
            for num in nums:
                if total < num:
                    break
                dp[total - num] += dp[total]
        return dp[0]
